sale_promotion_rule/models/sale_promotion_rule.py

```diff
@@ -103,6 +107,49 @@ class SalePromotionRule(models.Model):
 """,
     )
 
+    count_usage = fields.Integer(
+        string="Number of use",
+        help="Total number sale order which use this rule",
+        compute="_calc_count_usage",
+    )
+    budget_spent = fields.Monetary(
+        "Budget Spent", compute="_calc_budget_spent"
+    )
+    budget_max = fields.Monetary("Budget Max")
+
+    def _calc_count_usage(self):
+        for rec in self:
+            count = self.env["sale.order"].search_count(
+                [
+                    ("state", "in", ["sale", "done"]),
+                    "|",
+                    ("promotion_rule_ids", "in", rec.ids),
+                    ("coupon_promotion_rule_id", "=", rec.id),
+                ]
+            )
+            rec.count_usage = count
+
+    def _calc_budget_spent(self):
+        for rec in self:
+            lines = self.env["sale.order.line"].search(
+                [
+                    ("state", "in", ["sale", "done"]),
+                    ("promotion_rule_id", "=", rec.id),
+                    # "|",
+                    # ("promotion_rule_ids", "in", rec.ids),
+                    # ("coupon_promotion_rule_id", "=", rec.id),
+                ]
+            )
+            if lines:
+                if rec.discount_type == "amount_tax_included":
+                    rec.budget_spent = -sum(lines.mapped("price_total"))
+                else:
+                    rec.budget_spent = -sum(
+                        lines.mapped("untaxed_amount_to_invoice")
```

Comment:
It should be something else because untaxed_amount_to_invoice depends if the invoice has been generated or not (and if the goods have been delievered?).

Reply:
it's price_subtotal